Carbon.Domain.EntityFrameworkCore.Extensions/Repositories/TenantManagedEFRepositoryBase.cs

```diff
@@ -31,45 +31,44 @@ public TenantManagedEFRepositoryBase(DbContext ctx)
 
         public override async Task ConnectToSolution<T>(T relatedEntity)
         {
-            if (relatedEntity == null || relatedEntity.RelationalOwners == null || !relatedEntity.RelationalOwners.Any() || relatedEntity.RelationalOwners.Where(k => k == null).Any())
-                return;
-            bool relationsChanged = false;
-            
-            foreach (var ro in relatedEntity.RelationalOwners)
+			if (relatedEntity == null || relatedEntity.RelationalOwners.Where(k => k == null).Any())
+				return;
+
+			if (relatedEntity.RelationalOwners == null)
             {
-                ro.EntityId = relatedEntity.Id;
-                ro.EntityCode = relatedEntity.GetObjectTypeCode();
+				relatedEntity.RelationalOwners = new List<EntitySolutionRelation>();
             }
 
-            var oldRelations = await TargetErDbSet.Where(k => !k.IsDeleted && k.EntityCode == relatedEntity.GetObjectTypeCode() && k.EntityId == relatedEntity.Id).ToListAsync();
+			foreach (var ro in relatedEntity.RelationalOwners)
+			{
+				ro.EntityId = relatedEntity.Id;
+				ro.EntityCode = relatedEntity.GetObjectTypeCode();
+			}
 
-            if (oldRelations == null || !oldRelations.Any())
```

Comment:
buradaki if'i kaldırmam biraz kafa karışıklığı yaratabileceği için açıklayayım, artık yeni gelen dto'nun solution listesinin boş gelmesine izin verdiğimiz için, oldRelations boş veya null geldiğinde otomatikman relationların değiştiğini varsaymamız doğru olmayacaktır; çünkü dto'da hiç solution yoksa ve oldRelations da boş gelirse aslında relationsChanged false olmalıdır